pkg/generate/app/app.go

```diff
@@ -274,18 +296,27 @@ func (r *ImageRef) DeployableContainer() (container *kapi.Container, triggers []
 		if len(tag) == 0 {
 			tag = imageapi.DefaultImageTag
 		}
+		imageChangeParams := &deployapi.DeploymentTriggerImageChangeParams{
+			Automatic:      true,
+			ContainerNames: []string{name},
+			Tag:            tag,
+		}
+		if r.Stream != nil {
+			imageChangeParams.From = kapi.ObjectReference{
+				Kind:      "ImageStream",
+				Name:      r.Stream.Name,
+				Namespace: r.Stream.Namespace,
+			}
+		} else {
```

Comment:
hmm if you had a DockerImageRef, then no trigger should be generated. I'll add a check for that.